packages/techdocs-common/src/stages/publish/openStackSwift.ts

```diff
@@ -93,9 +95,13 @@ export class OpenStackSwiftPublish implements PublisherBase {
    * and the storage client will work.
    */
   getReadiness(): Promise<ReadinessResponse> {
-    return new Promise(resolve => {
-      this.storageClient.getContainer(this.containerName, (err, container) => {
-        if (container) {
+    return new Promise(async resolve => {
+      try {
+        const container = await this.storageClient.getContainerMetadata(
+          this.containerName,
+        );
+
+        if (!(container instanceof NotFound)) {
```

Comment:
This is an interesting API. Doesn't it throw an error in case of a failure / not found?

Reply:
no it's not, it returns a type called "NotFound"